NYPLAudiobookToolkit/Bookmarks/NYPLAudiobookBookmarkFactory.swift

```diff
@@ -1,7 +1,61 @@
 import Foundation
 import NYPLUtilities
 
+public protocol NYPLBookmarkSelectorParsing {
+  static func parseSelectorJSONString(fromServerAnnotation annotation: [String: Any],
+                                     annotationType: NYPLBookmarkSpec.Motivation,
+                                     bookID: String) -> String?
+}
+
 public final class NYPLAudiobookBookmarkFactory {
+  /// Factory method to create a new bookmark from a server annotation.
+  ///
+  /// - Parameters:
+  ///   - annotation: The annotation object coming from the server in a
+  ///   JSON-like structure.
+  ///   - annotationType: Whether it's an explicit bookmark or a reading progress.
+  ///   - bookID: The book the annotation is related to.
+  /// - Returns: a client-side representation of a bookmark.
+  public class func make(fromServerAnnotation annotation: [String: Any],
+                         selectorValueParser: NYPLBookmarkSelectorParsing.Type,
```

Comment:
This is a nice a solution! Maybe adding a bit about this parameter in the function documentation here might be helpful.